runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -685,13 +729,16 @@ impl<T: Trait> Module<T> {
 
             // immediately execute proposal if it ready for execution or save it for the future otherwise.
             if finalized_proposal.is_ready_for_execution(now) {
-                Self::execute_proposal(proposal_id);
+                executed_weight =
```

Comment:
Initial executed_weight = 0, you don't need to use `saturating_add` here, an assignment is enough.